src/test/libraryTest/type/typeUtils/typeUtils.ts

```diff
@@ -82,6 +82,26 @@ describe(`TypeUtils`, () => {
             expect(actual).deep.equal(expected);
         });
 
+        it(`simplify true | false to boolean`, () => {
+            const actual: ReadonlyArray<AbridgedType> = TypeUtils.simplify([
+                TypeUtils.logicalLiteralFactory(false, "true"),
+                TypeUtils.logicalLiteralFactory(false, "false"),
+            ]);
+            const expected: ReadonlyArray<AbridgedType> = [Type.LogicalInstance];
+            expect(actual).deep.equal(expected);
+        });
+
+        it(`prefer nullable in boolean simplification`, () => {
+            const actual: ReadonlyArray<AbridgedType> = TypeUtils.simplify([
+                TypeUtils.logicalLiteralFactory(false, "true"),
+                TypeUtils.logicalLiteralFactory(false, "false"),
+                TypeUtils.logicalLiteralFactory(true, "true"),
+                TypeUtils.logicalLiteralFactory(true, "false"),
```

Comment:
I don't understand this line.

Reply:
It's creating the 4 possible variants of a logical literal (flag for true/flase, and a flag for isNullable). The test is to see if `simplify(true | nullable false)` resolves to `nullable boolean`.

Reply:
Simplified lines by creating TrueInstance/FalseInstance/etc.